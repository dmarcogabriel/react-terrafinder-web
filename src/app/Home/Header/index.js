import React from 'react';
import { MdPhoneInTalk, MdMenu } from 'react-icons/md';
import { Link } from 'react-router-dom';
import Button from 'common/Button';
import LogoDesktop from './Logo.svg';
import Logo from './Logo.png';
import classes from './Header.module.scss';

import LINKS from './links';

export default function Header() {
  return (
    <div className={classes.header}>
      <img className={classes.logo} src={Logo} alt="logo" />

      <button type="button">
        <MdMenu className={classes.menu} />
      </button>

      <img
        className={classes.logoDesktop}
        alt="logo desktop"
        src={LogoDesktop}
      />

      {LINKS.map((link) => (
        <Link className={classes.link} key={link.key} to={link.href}>
          {link.name}
        </Link>
      ))}

      <Button className={classes.button}>
        <p className={classes.buttonContactText}>Ligue para n√≥s!</p>

        <div className={classes.buttonContactContent}>
          <MdPhoneInTalk className={classes.phoneIcon} />

          <p className={classes.buttonContactText}>+55 (14) 99999-9999</p>
        </div>
      </Button>
    </div>
  );
}
