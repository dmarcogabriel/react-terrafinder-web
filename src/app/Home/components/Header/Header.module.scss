@import 'styles/';

.header {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1000;
  background: $white;
  box-shadow: 1px 0 5px 5px rgba(0, 0, 0, 0.3);

  .content {
    position: relative;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: $size-sm $size-lg;

    .menuIcon {
      display: inline-block;
      font-size: 2.5rem;

      @media ($desktop) {
        display: none;
      }
    }

    .logoDesktop {
      height: 4rem;
      display: none;
      cursor: pointer;

      @media ($desktop) {
        display: inline-block;
      }
    }

    .logo {
      display: inline-block;
      height: 4rem;

      @media ($desktop) {
        display: none;
      }
    }

    .linkStyle {
      display: none;

      @media ($desktop) {
        display: inline-block;
        font-size: $size-md;
        color: $black;
      }
    }

    .button {
      display: none;
      flex-direction: column;
      padding: 1rem;
      background: $primary-dark;
      color: $white;

      @media ($desktop) {
        display: flex;
      }

      .buttonContactText {
        display: inline-block;
        font-size: $size-sm;
        padding: 0.25rem 0;
      }

      .buttonContactContent {
        display: flex;
        justify-content: center;
        align-items: center;

        .phoneIcon {
          font-size: 2rem;

          @media ($desktop) {
            font-size: 1.5rem;
          }
        }
      }
    }
  }
}

.menuList {
  position: fixed;
  display: flex;
  flex-direction: column;
  top: 0;
  left: 0;
  width: 100%;
  z-index: 500;
  background-color: $white;
  padding: 5.5rem 2rem 1rem;
  box-shadow: 1rem 0 1rem 1rem rgba(0, 0, 0, 0.3);

  .menuItem {
    font-size: $size-md;
    color: $black;
    margin: 1rem 0;
  }
}
