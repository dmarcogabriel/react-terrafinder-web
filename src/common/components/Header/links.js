export default [
  {
    href: '/home/search-property',
    name: 'Procurar Imóvel',
    key: 'l1',
  },
  {
    href: '/create/property',
    name: 'Quero anunciar meu imóvel',
    key: 'l2',
  },
  {
    href: '/home',
    name: 'Como funciona?',
    key: 'l3',
  },
];
