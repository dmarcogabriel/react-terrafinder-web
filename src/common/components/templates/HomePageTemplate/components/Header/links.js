export const LOGGED_LINKS = [
  {
    href: '/dashboard',
    name: 'Meus anúncios',
    key: 'l1',
  },
  {
    href: '/create/property',
    name: 'Quero anunciar meu imóvel',
    key: 'l2',
  },
  {
    href: '/home/search-property',
    name: 'Procurar Imóvel',
    key: 'l3',
  },
];

export const UNLOGGED_LINKS = [
  {
    href: '/search-property',
    name: 'Procurar Imóvel',
    key: 'l1',
  },
  {
    href: '/create/property',
    name: 'Quero anunciar meu imóvel',
    key: 'l2',
  },
];
