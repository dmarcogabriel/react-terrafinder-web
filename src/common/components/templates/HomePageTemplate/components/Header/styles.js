import styled, { css } from 'styled-components';
import Button from 'common/components/atm/Button';
import { MdPhoneInTalk } from 'react-icons/md';

export const ContactButton = styled(Button).attrs({ modifiers: 'primary' })`
  display: none;
  flex-direction: column;
  padding: 1rem;

  ${({ theme: { breakpoint } }) => css`
    @media (${breakpoint.desktop}) {
      display: flex;
    }
  `}

  > div {
    display: flex;
    justify-content: center;
    align-items: center;
  }
`;

export const ContactButtonText = styled.p`
  display: inline-block;
  font-size: 0.75rem;
  padding: 0.25rem 0;
`;

export const PhoneIcon = styled(MdPhoneInTalk)`
  font-size: 2rem;

  ${({ theme }) => css`
    @media (${theme.breakpoint.desktop}) {
      font-size: 1.5rem;
    }
  `}
`;

export const Logo = styled.img`
  height: 4rem;
  cursor: pointer;
`;
