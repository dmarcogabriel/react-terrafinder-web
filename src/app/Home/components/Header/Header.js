import React, { useState, useEffect } from 'react';
import { Link as RouterLink, useHistory } from 'react-router-dom';
import { useUser } from 'hooks/useUser';
import {
  AppBar,
  Toolbar,
  Button,
  Typography,
  IconButton,
  Menu,
  MenuItem,
} from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import logoImageDesktop from './static/Logo.svg';
import logoImage from './static/Logo.png';
import { LOGGED_LINKS, UNLOGGED_LINKS } from './links';
import { Logo } from './styles';
import { HeaderLink } from './HeaderLink';

export const Header = () => {
  const [showMenu, setShowMenu] = useState(null);
  const [links, setLinks] = useState([]);
  const { currentUser, logout, userIsLogged } = useUser();
  const history = useHistory();

  const handleLogout = () => {
    logout();
    history.replace('/home');
  };

  const handleShowMenu = (event) => setShowMenu(event.currentTarget);

  const handleCloseMenu = () => setShowMenu(null);

  const handleClickMenuItem = (url) => history.push(url);

  useEffect(() => {
    if (currentUser) {
      setLinks(LOGGED_LINKS);
    } else {
      setLinks(UNLOGGED_LINKS);
    }
  }, [currentUser]);

  return (
    <>
      <AppBar position="fixed" sx={{ backgroundColor: '#fff' }}>
        <Toolbar
          sx={{
            display: { xs: 'flex', md: 'none' },
            justifyContent: 'space-between',
          }}
        >
          <RouterLink to="/">
            <Logo alt="logo" src={logoImage} />
          </RouterLink>
          <IconButton size="large" onClick={handleShowMenu}>
            <MenuIcon />
          </IconButton>
        </Toolbar>
        <Toolbar
          sx={{
            display: { xs: 'none', md: 'flex' },
            justifyContent: 'space-between',
          }}
        >
          <Logo alt="logo desktop" src={logoImageDesktop} />
          {links.map((link) => (
            <RouterLink key={link.key} to={link.href} component={HeaderLink}>
              {link.name}
            </RouterLink>
          ))}
          {userIsLogged() ? (
            <>
              <Typography data-testid="greetings" sx={{ color: '#ed8936' }}>
                {`Ol치, ${currentUser.firstName}`}
              </Typography>
              <Button
                data-testid="logoutButton"
                onClick={handleLogout}
                sx={{ color: '#ed8936' }}
              >
                Logout
              </Button>
            </>
          ) : (
            <Button variant="contained" color="warning">
              <Typography>Fale conosco!!</Typography>
            </Button>
          )}
        </Toolbar>
      </AppBar>
      <Menu
        anchorEl={showMenu}
        open={Boolean(showMenu)}
        onClick={handleCloseMenu}
      >
        {links.map((link) => (
          <MenuItem
            key={link.key}
            onClick={() => handleClickMenuItem(link.href)}
          >
            {link.name}
          </MenuItem>
        ))}
        {userIsLogged() ? (
          <>
            <MenuItem>
              <Typography data-testid="greetings" sx={{ color: '#ed8936' }}>
                {`Ol치, ${currentUser.firstName}`}
              </Typography>
            </MenuItem>
            <MenuItem>
              <Button
                data-testid="logoutButton"
                onClick={handleLogout}
                sx={{ color: '#ed8936' }}
              >
                Logout
              </Button>
            </MenuItem>
          </>
        ) : (
          <MenuItem>
            <Button variant="contained" color="warning">
              <Typography>Fale conosco!!</Typography>
            </Button>
          </MenuItem>
        )}
      </Menu>
    </>
  );

  // return (
  //   <>
  //     <div className={classes.header}>
  //       <div className={classes.content}>
  //         <Link to="/home" className={classes.logo}>
  //           <img src={Logo} alt="logo" />
  //         </Link>

  //         <button
  //           data-testid="menuButton"
  //           type="button"
  //           onClick={() => setShowMenu(!showMenu)}
  //         >
  //           <MdMenu className={classes.menuIcon} />
  //         </button>

  //         <Link to="/home" className={classes.logoDesktop}>
  //           <img alt="logo desktop" src={LogoDesktop} />
  //         </Link>

  //         {links.map((link) => (
  //           <Link className={classes.linkStyle} key={link.key} to={link.href}>
  //             {link.name}
  //           </Link>
  //         ))}

  //         {currentUser ? (
  //           <>
  //             <p
  //               data-testid="greetings"
  //               className={classes.grettings}
  //             >{`Ol치, ${currentUser.firstName}`}</p>

  //             <button
  //               data-testid="logoutButton"
  //               type="button"
  //               className={classes.logout}
  //               onClick={handleLogout}
  //             >
  //               Logout
  //             </button>
  //           </>
  //         ) : (
  //           <ContactButton>
  //             <ContactButtonText>Ligue para n칩s!</ContactButtonText>

  //             <div>
  //               <PhoneIcon />

  //               <ContactButtonText>+55 (14) 99999-9999</ContactButtonText>
  //             </div>
  //           </ContactButton>
  //         )}
  //       </div>
  //     </div>

  //     {showMenu && (
  //       <div data-testid="menu" className={classes.menuList}>
  //         {links.map((link) => (
  //           <Link className={classes.menuItem} key={link.key} to={link.href}>
  //             {link.name}
  //           </Link>
  //         ))}

  //         <button
  //           type="button"
  //           className={classes.logoutLink}
  //           onClick={handleLogout}
  //         >
  //           Logout
  //         </button>
  //       </div>
  //     )}
  //   </>
  // );
};
