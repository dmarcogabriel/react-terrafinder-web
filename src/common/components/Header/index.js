import React, { useState, useEffect } from 'react';
import { MdMenu } from 'react-icons/md';
import { Link, useHistory } from 'react-router-dom';
import { useUser } from 'hooks/useUser';
import LogoDesktop from '../../static/Logo.svg';
import Logo from './Logo.png';
import classes from './Header.module.scss';
import { LOGGED_LINKS, UNLOGGED_LINKS } from './links';
import { ContactButton, ContactButtonText, PhoneIcon } from './styles';

export default function Header() {
  const [showMenu, setShowMenu] = useState(false);
  const [links, setLinks] = useState([]);
  const { currentUser, logout } = useUser();
  const history = useHistory();

  const handleLogout = () => {
    logout();
    history.replace('/home');
  };

  useEffect(() => {
    if (currentUser) {
      setLinks(LOGGED_LINKS);
    } else {
      setLinks(UNLOGGED_LINKS);
    }
  }, [currentUser]);

  return (
    <>
      <div className={classes.header}>
        <div className={classes.content}>
          <Link to="/home" className={classes.logo}>
            <img src={Logo} alt="logo" />
          </Link>

          <button
            data-testid="menuButton"
            type="button"
            onClick={() => setShowMenu(!showMenu)}
          >
            <MdMenu className={classes.menuIcon} />
          </button>

          <Link to="/home" className={classes.logoDesktop}>
            <img alt="logo desktop" src={LogoDesktop} />
          </Link>

          {links.map((link) => (
            <Link className={classes.linkStyle} key={link.key} to={link.href}>
              {link.name}
            </Link>
          ))}

          {currentUser ? (
            <>
              <p
                data-testid="greetings"
                className={classes.grettings}
              >{`Olá, ${currentUser.firstName}`}</p>

              <button
                data-testid="logoutButton"
                type="button"
                className={classes.logout}
                onClick={handleLogout}
              >
                Logout
              </button>
            </>
          ) : (
            <ContactButton>
              <ContactButtonText>Ligue para nós!</ContactButtonText>

              <div>
                <PhoneIcon />

                <ContactButtonText>+55 (14) 99999-9999</ContactButtonText>
              </div>
            </ContactButton>
          )}
        </div>
      </div>

      {showMenu && (
        <div data-testid="menu" className={classes.menuList}>
          {links.map((link) => (
            <Link className={classes.menuItem} key={link.key} to={link.href}>
              {link.name}
            </Link>
          ))}

          <button
            type="button"
            className={classes.logoutLink}
            onClick={handleLogout}
          >
            Logout
          </button>
        </div>
      )}
    </>
  );
}
