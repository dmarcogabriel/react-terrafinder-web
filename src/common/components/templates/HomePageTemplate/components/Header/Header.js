import React, { useState, useEffect } from 'react';
import { Link as RouterLink, useHistory } from 'react-router-dom';
import { useUser } from 'hooks/useUser';
import {
  AppBar,
  Toolbar,
  Button,
  Typography,
  IconButton,
  Menu,
  MenuItem,
} from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import logoImageDesktop from './static/Logo.svg';
import logoImage from './static/Logo.png';
import { LOGGED_LINKS, UNLOGGED_LINKS } from './links';
import { Logo } from './styles';
import { HeaderLink } from './HeaderLink';

export default function Header() {
  const [showMenu, setShowMenu] = useState(null);
  const [links, setLinks] = useState([]);
  const { currentUser, logout, userIsLogged } = useUser();
  const history = useHistory();

  const handleLogout = () => {
    logout();
    history.replace('/');
  };

  const handleShowMenu = (event) => setShowMenu(event.currentTarget);

  const handleCloseMenu = () => setShowMenu(null);

  const handleClickMenuItem = (url) => history.push(url);

  const handleClickContactUs = () => history.push('contact-us');

  useEffect(() => {
    if (currentUser) {
      setLinks(LOGGED_LINKS);
    } else {
      setLinks(UNLOGGED_LINKS);
    }
  }, [currentUser]);

  return (
    <>
      <AppBar
        data-testid="app-bar"
        position="fixed"
        sx={{ backgroundColor: '#fff' }}
      >
        <Toolbar
          sx={{
            display: { xs: 'flex', md: 'none' },
            justifyContent: 'space-between',
          }}
        >
          <RouterLink to="/">
            <Logo alt="logo" src={logoImage} />
          </RouterLink>
          <IconButton
            data-testid="menuButton"
            size="large"
            onClick={handleShowMenu}
          >
            <MenuIcon />
          </IconButton>
        </Toolbar>
        <Toolbar
          sx={{
            display: { xs: 'none', md: 'flex' },
            justifyContent: 'space-between',
          }}
        >
          <RouterLink to="/">
            <Logo alt="logo desktop" src={logoImageDesktop} />
          </RouterLink>
          {links.map((link) => (
            <RouterLink key={link.key} to={link.href} component={HeaderLink}>
              {link.name}
            </RouterLink>
          ))}
          {userIsLogged() ? (
            <>
              <Typography data-testid="greetings" sx={{ color: '#ed8936' }}>
                {`Olá, ${currentUser.firstName}`}
              </Typography>
              <Button
                data-testid="logoutButton"
                onClick={handleLogout}
                sx={{ color: '#ed8936' }}
              >
                Logout
              </Button>
            </>
          ) : (
            <Button
              variant="contained"
              color="warning"
              onClick={handleClickContactUs}
            >
              <Typography>Fale conosco!!</Typography>
            </Button>
          )}
        </Toolbar>
      </AppBar>
      <Menu
        data-testid="menu"
        anchorEl={showMenu}
        open={Boolean(showMenu)}
        onClick={handleCloseMenu}
      >
        {links.map((link) => (
          <MenuItem
            key={link.key}
            data-testid={`menu-item-${link.key}`}
            onClick={() => handleClickMenuItem(link.href)}
          >
            {link.name}
          </MenuItem>
        ))}
        {userIsLogged() && (
          <MenuItem>
            <Typography data-testid="greetings" sx={{ color: '#ed8936' }}>
              {`Olá, ${currentUser.firstName}`}
            </Typography>
          </MenuItem>
        )}
        {userIsLogged() ? (
          <MenuItem>
            <Button
              data-testid="logoutButton"
              onClick={handleLogout}
              sx={{ color: '#ed8936' }}
            >
              Logout
            </Button>
          </MenuItem>
        ) : (
          <MenuItem>
            <Button variant="contained" color="warning">
              <Typography>Fale conosco!</Typography>
            </Button>
          </MenuItem>
        )}
      </Menu>
    </>
  );
}
