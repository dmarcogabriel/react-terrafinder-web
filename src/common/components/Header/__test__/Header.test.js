import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import Header from '..';

let replaceResponse;
const mockReplace = jest.fn((url) => {
  replaceResponse = url;
});

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    replace: mockReplace,
  }),
  Link: ({ to, children }) => <a href={to}>{children}</a>,
}));

let mockUser;
const mockLogout = jest.fn();

jest.mock('hooks/useUser', () => ({
  useUser: () => ({
    currentUser: mockUser,
    logout: mockLogout,
  }),
}));

describe('<Header />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('logged', () => {
    mockUser = { firstName: 'Tester' };

    it('should show grettings', () => {
      const { getByTestId } = render(<Header />);

      const greetings = getByTestId('greetings');
      expect(greetings).toBeInTheDocument();
      expect(greetings).toHaveTextContent('OlÃ¡, Tester');
    });

    it('should show logout button', () => {
      const { getByTestId } = render(<Header />);

      const logoutButton = getByTestId('logoutButton');
      expect(logoutButton).toBeInTheDocument();
      expect(logoutButton).toHaveTextContent('Logout');
    });

    it('should show menu', () => {
      const { getByTestId } = render(<Header />);

      const menuButton = getByTestId('menuButton');
      fireEvent.click(menuButton);

      const menu = getByTestId('menu');
      expect(menu).toBeInTheDocument();
    });

    it('should handle logout', () => {
      const { getByTestId } = render(<Header />);

      const logoutButton = getByTestId('logoutButton');
      fireEvent.click(logoutButton);

      expect(mockLogout).toHaveBeenCalled();
      expect(mockReplace).toHaveBeenCalled();
      expect(replaceResponse).toEqual('/home');
    });
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
