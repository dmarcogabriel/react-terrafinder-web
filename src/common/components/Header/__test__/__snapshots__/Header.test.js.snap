// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/components/Header matches snapshot 1`] = `
<div>
  <div>
    <a
      href="/home"
    >
      <img
        alt="logo"
        src="\\"Logo.png\\""
      />
    </a>
    <button
      data-testid="menuButton"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
    <a
      href="/home"
    >
      <img
        alt="logo desktop"
        src="\\"Logo.svg\\""
      />
    </a>
    <p
      data-testid="greetings"
    >
      Ol√°, Tester
    </p>
    <button
      data-testid="logoutButton"
      onClick={[Function]}
      type="button"
    >
      Logout
    </button>
  </div>
</div>
`;
